Error: couldn't find key postgresql-password in Secret

### Issue submitter TODO list

- [X] I've searched for an already existing issues [here](https://github.com/sentry-kubernetes/charts/issues)

### Describe the bug (actual behavior)
```
k get pod  -n test | grep CreateContainerConfigError
sentry-cron-5986567d77-t2kpm       0/1     CreateContainerConfigError   0          64s
sentry-web-75dd588596-q7sp8        0/1     CreateContainerConfigError   0          64s
sentry-worker-5bdb4445f8-wnbxq     0/1     CreateContainerConfigError   0          64s
```

Events
```
k get event -n test | grep Error
3s          Warning   Failed                             pod/sentry-cron-5986567d77-t2kpm                                            Error: couldn't find key postgresql-password in Secret test/my-postgresql
20m         Warning   Unhealthy                          pod/sentry-rabbitmq-0                                                       Readiness probe failed: Error:...
19m         Warning   Failed                             pod/sentry-snuba-outcomes-consumer-6c4bf67d4b-gjmqn                         Error: secret "sentry-snuba-env" not found
20m         Warning   Failed                             pod/sentry-snuba-subscription-consumer-transactions-77f6d8b7c4zzvhk         Error: secret "sentry-snuba-env" not found
0s          Warning   Failed                             pod/sentry-web-75dd588596-q7sp8                                             Error: couldn't find key postgresql-password in Secret test/my-postgresql
6s          Warning   Failed                             pod/sentry-worker-5bdb4445f8-wnbxq                                          Error: couldn't find key postgresql-password in Secret test/my-postgresql
```


### Expected behavior

install sentry helm chart without error

### Your installation details
```
postgresql:
  enabled: false

externalPostgresql:
  host: my-postgresql.test.svc.cluster.local
  port: 5432
  username: postgres
  existingSecret: my-postgresql
  database: postgres
```
### Steps to reproduce
```
helm install -n test my-postgresql bitnami/postgresql
write values.yaml
helm install -n test --wait sentry . --values values.yaml --timeout=1000s
```
### Screenshots

_No response_

### Logs
```
k get secret -n test my-postgresql -o yaml
apiVersion: v1
data:
  postgres-password: d29xRUJnNkN5Uw==
kind: Secret
metadata:
  annotations:
    meta.helm.sh/release-name: my-postgresql
    meta.helm.sh/release-namespace: test
  creationTimestamp: "2024-09-07T11:55:17Z"
  labels:
    app.kubernetes.io/instance: my-postgresql
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: postgresql
    app.kubernetes.io/version: 16.4.0
    helm.sh/chart: postgresql-15.5.28
  name: my-postgresql
  namespace: test
  resourceVersion: "893340"
  uid: 94a997b2-4a53-4cf0-b72e-4236dd956d9b
type: Opaque
```



### Additional context

_No response_

show secret with key postgresql-password

```
k get secret -n test my-postgresql -o yaml
apiVersion: v1
data:
  postgresql-password: bUt3R1c5eGlsMA==
kind: Secret
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","data":{"postgresql-password":"bUt3R1c5eGlsMA=="},"kind":"Secret","metadata":{"annotations":{"meta.helm.sh/release-name":"my-postgresql","meta.helm.sh/release-namespace":"test"},"creationTimestamp":"2024-09-07T14:34:51Z","labels":{"app.kubernetes.io/instance":"my-postgresql","app.kubernetes.io/managed-by":"Helm","app.kubernetes.io/name":"postgresql","app.kubernetes.io/version":"16.4.0","helm.sh/chart":"postgresql-15.5.28"},"name":"my-postgresql","namespace":"test","resourceVersion":"948000","uid":"e500c377-d044-4b45-a7bb-bf7cbd662da9"},"type":"Opaque"}
    meta.helm.sh/release-name: my-postgresql
    meta.helm.sh/release-namespace: test
  creationTimestamp: "2024-09-07T14:37:11Z"
  labels:
    app.kubernetes.io/instance: my-postgresql
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: postgresql
    app.kubernetes.io/version: 16.4.0
    helm.sh/chart: postgresql-15.5.28
  name: my-postgresql
  namespace: test
  resourceVersion: "948694"
  uid: 82e0e45b-81e2-4892-a4ac-e0561e3bc5bb
type: Opaque
```


Try install with values.yaml
```
postgresql:
  enabled: false
externalPostgresql:
  host: my-postgresql.test.svc.cluster.local
  port: 5432
  username: postgres
  existingSecret: my-postgresql
  database: postgres
```
error:

```
helm install -n test --wait sentry . --values values.yaml --timeout=1000s
coalesce.go:237: warning: skipped value for kafka.config: Not a table.
Error: INSTALLATION FAILED: template: sentry/templates/sentry/subscription-consumer/transactions/deployment-sentry-subscription-consumer-transactions.yaml:94:3: executing "sentry/templates/sentry/subscription-consumer/transactions/deployment-sentry-subscription-consumer-transactions.yaml" at <include "sentry.env" .>: error calling include: template: sentry/templates/_helper.tpl:518:24: executing "sentry.env" at <.Values.externalPostgresql.existingSecretKeys.password>: nil pointer evaluating interface {}.password
```

--- a/charts/sentry/values.yaml
+++ b/charts/sentry/values.yaml
@@ -2056,7 +2056,8 @@ externalPostgresql:
   username: postgres
   # password: postgres
   # existingSecret: secret-name
-  ## If externalPostgresql.existingSecret is used, externalPostgresql.existingSecretKeys.password must be specified.
+  # set existingSecretKeys in a secret, if not specified, value from the secret won't be used
+  # if externalPostgresql.existingSecret is used, externalPostgresql.existingSecretKeys.password must be specified.
   existingSecretKeys: {}
   #   password: password # Required if existingSecret is used.
   #   username: username

--- a/charts/sentry/values.yaml
+++ b/charts/sentry/values.yaml
@@ -2059,7 +2059,7 @@ externalPostgresql:
   # set existingSecretKeys in a secret, if not specified, value from the secret won't be used
   # if externalPostgresql.existingSecret is used, externalPostgresql.existingSecretKeys.password must be specified.
   existingSecretKeys: {}
-  #   password: postgresql-password # Required if existingSecret is used.
+  #   password: postgresql-password # Required if existingSecret is used. Key in existingSecret.
   #   username: username
   #   database: database
   #   port: port

--- a/charts/sentry/values.yaml
+++ b/charts/sentry/values.yaml
@@ -2059,7 +2059,7 @@ externalPostgresql:
   # set existingSecretKeys in a secret, if not specified, value from the secret won't be used
   # if externalPostgresql.existingSecret is used, externalPostgresql.existingSecretKeys.password must be specified.
   existingSecretKeys: {}
-  #   password: password # Required if existingSecret is used.
+  #   password: postgresql-password # Required if existingSecret is used.
   #   username: username
   #   database: database
   #   port: port

--- a/charts/sentry/values.yaml
+++ b/charts/sentry/values.yaml
@@ -2050,16 +2050,15 @@ postgresql:
 
 ## This value is only used when postgresql.enabled is set to false
 ## Set either externalPostgresql.password or externalPostgresql.existingSecret to configure password
-## externalPostgresql.existingSecret should have a key of 'postgresql-password' which holds the password
 externalPostgresql:
   # host: postgres
   port: 5432
   username: postgres
   # password: postgres
   # existingSecret: secret-name
-  ## set existingSecretKeys in a secret, if not specified, value from the secret won't be used
+  ## If externalPostgresql.existingSecret is used, externalPostgresql.existingSecretKeys.password must be specified.
   existingSecretKeys: {}
-  #   password: password
+  #   password: password # Required if existingSecret is used.
   #   username: username
   #   database: database
   #   port: port

